lib/shopping_cart.rb

```diff
@@ -0,0 +1,45 @@
+class ShoppingCart
+
+  attr_reader :name, :capacity, :products, :details
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.chomp("items").to_i
+    @products = []
+    @details_hash = {}
+    @sum = 0
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    @details_hash[:name] = @name
+    @details_hash[:capacity] = @capacity
+    @details_hash
```

Comment:
I would take the details_hash out of initialize because in order for it to get filled, this details method needs to get called. I think in this method you could jsut do something like: 
{name: @name, capacity: @capacity}